back/engines/commercial/multi_tenancy/lib/tasks/core/restore_s3_objects.rake

```diff
@@ -0,0 +1,50 @@
+# frozen_string_literal: true
+
+# This task should be used if partial data has been restored to a schema from a backup DB
+# to renable the S3 file assets that were flagged as deleted when the data was deleted.
+# Currently only restores assets associated with Ideas (IdeaImage, IdeaFile, TextFile).
+
+namespace :fix_existing_tenants do
+  desc 'If ideas have been manually restored from backup - restore any deleted images.'
+  task :restore_s3_objects, [:schema] => [:environment] do |_t, args|
+    schema = args[:schema]
+    unless schema
+      puts 'Please provide a tenant schema name to restore the images for.'
+      exit
+    end
+
+    Apartment::Tenant.switch(schema) do
+      Rails.logger.info "STARTED: Finding delete markers for #{schema}"
+      bucket = ENV.fetch('AWS_S3_BUCKET', nil)
```

Comment:
I guess we can make it already fail at this stage if `AWS_S3_BUCKET` is missing.
```suggestion
      bucket = ENV.fetch('AWS_S3_BUCKET')
```